compiler/ilgen/OMRIlBuilder.cpp

```diff
@@ -2545,26 +2545,111 @@ OMR::IlBuilder::Switch(const char *selectionVar,
    TR::Node *defaultNode = TR::Node::createCase(0, (*defaultBuilder)->getEntry()->getEntry());
    TR::Node *lookupNode = TR::Node::create(TR::lookup, numCases + 2, loadValue(selectorValue), defaultNode);
 
+   generateSwitchCases(lookupNode, defaultNode, defaultBuilder, numCases, cases);
+   }
+
+void
+OMR::IlBuilder::Switch(const char *selectionVar,
+                  TR::IlBuilder **defaultBuilder,
+                  uint32_t numCases,
+                  ...)
+   {
+   va_list args;
+   va_start(args, numCases);
+   JBCase **cases = createCaseArray(numCases, args);
+   va_end(args);
+
+   Switch(selectionVar, defaultBuilder, numCases, cases);
+   }
+
+void
+OMR::IlBuilder::TableSwitch(const char *selectionVar,
+                  TR::IlBuilder **defaultBuilder,
+                  bool generateBoundsCheck,
+                  uint32_t numCases,
+                  JBCase **cases)
+   {
+   TR::IlValue *selectorValue = Load(selectionVar);
+   TR_ASSERT(selectorValue->getDataType() == TR::Int32, "TableSwitch only supports selector having type Int32");
+   if (numCases > 0)
+      {
+      int32_t low = cases[0]->_value;
+      int32_t high = cases[numCases - 1]->_value;
+      if (low != 0)
+         selectorValue = Sub(selectorValue, ConstInt32(low));
```

Comment:
Doesn't look like you use `high`?

Reply:
Yep!  I was using it but removed the use. I will remove it.